src/Core/FileSystemModule/IETempFile.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+//internal dependencies
+#include "IEFile.hpp"
+
+// std dependencies
+#include <vector>
+#include <string>
+#include <fstream>
+
+/**
+ * A collection of data stored in RAM
+ */
+
+class IETempFile {
+public:
+    std::string name{};
+    std::vector<char> data{}; // the data in the file
+    std::string extension{};
+
+    explicit IETempFile(const std::string& fileName, const std::vector<char>& fileData = {}) {
```

Comment:
Each of the functions here should be split between the IETempFile.hpp and IETempFile.cpp files.